build_product_images.py

```diff
@@ -149,8 +192,164 @@ def product_to_build(product_name, product_version, products):
+    args_dummy = copy.deepcopy(args)
+
+    if not rust_builder:
+        print("Building rust builder")
+        subprocess.run("make")
+
+    if not java:
+        args_dummy.image_version = '0'
+        args_dummy.product_version = '11'
+        print('Building dependency Java-Base', args_dummy.product_version)
+
+        run_commands(args_dummy.dry, build_and_publish_image(args_dummy, product_to_build('java-base', args_dummy.product_version, products)))
+
+        args_dummy.image_version = '0'
+        args_dummy.product_version = '1.8.0'
+        print('Building dependency Java-Base', args_dummy.product_version)
+
+        run_commands(args_dummy.dry, build_and_publish_image(args_dummy, product_to_build('java-base', args_dummy.product_version, products)))
+
+    if not tools:
+        args_dummy.image_version = '0'
+        args_dummy.product_version = '0.2.0'
+        print("Building dependency Tools", args_dummy.product_version)
+
+        run_commands(args_dummy.dry, build_and_publish_image(args_dummy, product_to_build('tools', args_dummy.product_version, products)))
+
+
+def check_platform(architecture):
+    """
+    Checks if a desired platform is given, gives current platform if not
+    """
+    if architecture is None:
+        architecture = platform.machine()
+
+    return architecture
+
+
+def create_virtual_enviroment(args):
+
+    commands = []
+
+    commands.append(
+        [
+            "docker",
+            "buildx",
+            "create",
+            "--name",
+            "builder",
+            "--use"
+        ]
+    )
+    run_commands(args.dry, commands)
+
+
+def remove_virtual_enviroment(args):
+
+    commands = []
+    commands.append(
+        [
+            "docker",
+            "buildx",
+            "rm",
+            "builder"
+        ]
+    )
+    run_commands(args.dry, commands)
+
+
+def join_nodes(args):
+
+    nodes = args.node.split(',')
+
+    for node in nodes:
+        commands = []
+        if node != nodes[0]:
+            commands.append(
+                [
+                    "docker",
+                    "buildx",
+                    "create",
+                    "--name",
+                    nodes[0],
+                    "--append",
+                    node
+                ]
+            )
+            run_commands(args.dry, commands)
+
+    use_builder(args)
+
+
+def use_builder(args):
+
+    nodes = args.node.split(',')
+    commands = []
+
+    commands.append(
+        [
+            "docker",
+            "buildx",
+            "use",
+            nodes[0]
+        ]
+    )
+
+    run_commands(args.dry, commands)
+
+
 def main():
     args = parse_args()
+    print("Current Platform: ", platform.machine())
+
+    if args.check:
+        check_or_build_dependencies(args, check_platform(args.architecture), conf.products)
+
+    if args.multiarch:
+        if args.node is None:
+            create_virtual_enviroment(args)
+
+    if args.node is not None:
+        join_nodes(args)
```

Comment:
Again, I never used `buildx` ... is this some form of speedup for local builds ? Because it's not used in the GH action. Is it really needed?

Reply:
This because if you build multi arch and give docker --platform with at least two target platforms, it needs a different builder to be happy. Currently we do not have native worker for arm64, therefore in the beginning the emulated for arm64 will be used. 
Due to poor performance, I suspect us using native nodes at some point. That's meant to be the preparation for building multi arch with native nodes.

Reply:
Done, since we do not use native worker right now.